trait/web-service.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  ContainerPort,
+  Deployment,
+  merge,
+  Service,
+  ServicePort,
+} from "../mod.ts";
+
+export class WebService {
+  deployment: Deployment;
+  service: Service;
+
+  constructor(webService: string, image: string, replica: number) {
+    const labels = { app: webService };
+    this.deployment = {
+      apiVersion: "apps/v1",
+      kind: "Deployment",
+      metadata: {
+        name: webService,
+        labels: labels,
+      },
+      spec: {
+        selector: { matchLabels: { app: webService } },
+        replicas: replica,
+        template: {
+          metadata: {
+            labels: labels,
+          },
+          spec: {
+            containers: [{
+              name: webService,
+              image: image,
+            }],
+          },
+        },
+      },
+    };
+
+    this.service = {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        name: webService,
+      },
+      spec: {
+        clusterIP: "None",
+        selector: { app: webService },
+      },
+    };
+  }
+
+  withContainerPorts(
```

Comment:
This method directly mutates the field `this.deployment`, so I think it's better to prefix the name with `set`, `with` kind of reminds me that it's immutable.

Reply:
right